spec/unit/environment/name_spec.rb

```diff
@@ -149,6 +165,8 @@
         it "replaces invalid characters in #{branch} with underscores" do
           bn = described_class.new(branch.dup, {:correct => true})
           expect(bn.dirname).to eq branch.gsub(/\W/, '_')
+          expect(bn.name).to eq branch
```

Comment:
huh, I would have thought this would also get corrected, but I guess I'm not totally sure where all the `name` is used.

Reply:
Yeah. After taking a look through it, I realized that it's currently inconsistent. The original implementation of `strip_component` introduced the inconsistency. I'll take responsibility for that. There would ideally be a clear separation between the input value given to `R10K::Environment::Name` and the derived code-environment name on disk (currently called `#dirname`). Bit of work to clean it all up at this point.

Reply:
ahh. Do you think it's a worthwhile bit of work to do eventually? If so, would you mind opening a ticket for it?